### ISSUE TITLE
Add JupyterLab UI tests

### ISSUE THREAD

**Issue by manics** [jupyterhub/jupyter-server-proxy#268]
We have good test coverage for the server component of this project, but nothing on the UI side. We should add something for JupyterLab since that seems to be where most development is occurring.

**Comment by jtpio** [comment]
Maybe using `galata`? 

https://github.com/jupyterlab/galata

**Comment by manics** [comment]
@bollwyvl has opened https://github.com/jupyterhub/jupyter-server-proxy/pull/269, I haven't had time to review it yet.

Related, @bollwyvl I've checked with the JupyterHub team and we'd like to invite you to be a maintainer of this repo, since we don't have much JupyterLab experience and you seem keen 😀.

**Comment by jtpio** [comment]
Nice :+1: 

#269 looks good, and covers the classic notebook too.

**Comment by bollwyvl** [comment]
Yeah, as has been mentioned, galata's lab-only focus is a bit of a glass cannon a) once it gets off the Lab path and b) the team looking to own it as a dependency hasn't done a lot of node/jest work and/or c) the scope is beyond what galata thinks would be needed to test features. 

I wrote `robotframework-jupyterlibrary` specifically for the case to be installed next to _whatever_ jupyter client (as long as it's written in such a way as to _be_ testable) without _requiring_ the other ones, as well as handling (multiple) server/remote components for testing collaborative features, and gathering plain-language test descriptions from the non-first-estate of js developers.

For example, on #269, we could add some test teardown invariants that would be like:

```robot
*** Settings ***
Library        ./logs.py
Test Teardown  Server Log Should Not Contain Tracebacks
```

And while one _could_ write all the I/O in robot, that case might be best suited to dropping to python, and could be a function called `def server_log_should_not_contain_tracebacks()`. If one really _does_ like writing JS, it can just punch that to the browser with `Execute JavaScript`, with or without `await`.

While robot itself is solid, [seleniumlibrary](https://robotframework.org/SeleniumLibrary/SeleniumLibrary.html) is still a bit of a bear... and the install toolchain for the successor based on playwright seems... complex. grpcio and node and another bespoke browser as a dependency of a python package are basically my personal packaging nightmare. 

However they are run, I've found keeping the scope of full, end-to-end acceptance tests to a few key workflows run in _as real an environment as possible_, and then leveraging the _hell_ out of the outputs (use the screenshots in docs, keeping the machine-readable ones over time, etc.) can bear the most fruit, while leveraging closer-to-the-engineer expertise for unit (with mocks) and integration (everything but a browser and The Internet) tests.

**Comment by jtpio** [comment]
Yes having a Galata example in https://github.com/jupyterlab/extension-examples would be useful for sure.

But for `jupyter-server-proxy` https://github.com/jupyterhub/jupyter-server-proxy/pull/269 should be sufficient for now.
